src/test/java/com/googleinterns/zoomtube/servlets/LectureServletTest.java

```
@@ -0,0 +1,151 @@
+    when(request.getParameter(LINK_INPUT)).thenReturn(TEST_LINK);
+    datastoreService.put(servlet.createLectureEntity(request));
+    
+    servlet.doPost(request, response);
+    
+    assertThat(datastoreService.prepare(new Query(Lecture.ENTITY_KIND)).countEntities()).isEqualTo(1);
+    verify(response).sendRedirect("/lecture-view.html?id=1&video-id=wXhTHyIgQ_U");
+  }
+
+  @Test
+  public void doPost_urlNotInDatabase_shouldAddToDatabaseAndReturnRedirect() throws IOException {
+    when(request.getParameter(LINK_INPUT)).thenReturn(TEST_LINK);
+    
+    // No lecture in datastoreService.
+    servlet.doPost(request, response);
+    
+    assertThat(datastoreService.prepare(new Query(Lecture.ENTITY_KIND)).countEntities()).isEqualTo(1);
+    verify(response).sendRedirect("/lecture-view.html?id=1&video-id=wXhTHyIgQ_U");
+  }
+
+  @Test
+  public void doGet_emptyDatabase_shouldReturnNoLecture() throws IOException {
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+    
+    String json = content.toString();
+    assertThat(json).startsWith("[]");
+  }
+
+  @Test
+  public void doGet_oneLectureInDatabase_shouldReturnOneLecture() throws IOException {
+    when(request.getParameter(LINK_INPUT)).thenReturn(TEST_LINK);
+    datastoreService.put(servlet.createLectureEntity(request));
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    String json = content.toString();
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    Type listType = new TypeToken<ArrayList<Lecture>>() {}.getType();
+    ArrayList<Lecture> lectures = gson.fromJson(json, listType);
+    assertThat(lectures.get(0).videoUrl()).isEqualTo(TEST_LINK);
```

I would advise that we do ```assertThat(lectures).hasSize(1)``` first to prevent a potential NullPointerException. It would also fail early and you would know the issue right away.